src/lib.rs

```diff
@@ -158,24 +158,45 @@ mod getifaddrs_posix {
     use std::ffi::CStr;
     use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 
+    // Wrapper around a sockaddr pointer. Guaranteed to not be null.
+    pub struct SockAddr {
+        inner: *const posix_sockaddr,
```

Comment:
Sorry for taking so long to look at this. I completely forgot about it.

I don't really like that this carries a raw pointer. It should either use a reference or just carry the `posix_sockaddr` directly inline. Alternatively, why not just have a function to construct a `std::net::SocketAddr` directly from a `*const posix_sockaddr`.

Reply:
> I don't really like that this carries a raw pointer. It should either use a reference or just carry the posix_sockaddr directly inline.

Agreed, I improved this part of the code.

> Alternatively, why not just have a function to construct a std::net::SocketAddr directly from a *const posix_sockaddr.

We can't use `SocketAddr` because it requires a port number, but I added a function to return `Option<IpAddr>` directly from `*const posix_sockaddr`. This also simplified the code a lot!

Let me know how this looks as I still have to make the equivalent changes in the Windows-specific portion of the code.